frontend/src/tests/instructor-permission-test.js

```diff
@@ -516,51 +469,254 @@ export function instructorsPermissionTests(api) {
+            expect(resp).toHaveStatus("success");
+            instructor = resp.payload.find(
+                (instructor) => instructor.utorid === instructorUser.utorid
+            );
+            expect(instructor).toBeDefined();
+
+            // Fetch the position related to our instructor
+            resp = await apiGET(`/admin/sessions/${session.id}/positions`);
+            position = resp.payload.find((position) =>
+                position.instructor_ids.find((id) => id === instructor.id)
+            );
+            expect(position).toBeDefined();
+
+            // Create another assignment (with no DDAH)
+            // for position associated with our instructor
+            const newAssignment = {
+                note: "",
+                applicant_id: databaseSeeder.seededData.applicant.id,
+                position_id: position.id,
+                start_date: "2019-09-02T00:00:00.000Z",
+                end_date: "2019-12-31T00:00:00.000Z",
+            };
+            resp = await apiPOST("/admin/assignments", newAssignment);
+            expect(resp).toHaveStatus("success");
+            assignment = resp.payload;
+
+            // Fetch the position related to other instructor
+            resp = await apiGET(`/admin/instructors`);
+            const otherInstructor = resp.payload.find(
+                (instructor) =>
+                    instructor.utorid ===
+                    databaseSeeder.seededData.instructors[4].utorid
+            );
+            expect(otherInstructor).toBeDefined();
+            resp = await apiGET(`/admin/sessions/${session.id}/positions`);
+            const otherPosition = resp.payload.find((position) =>
+                position.instructor_ids.find((id) => id === otherInstructor.id)
+            );
+            expect(otherPosition).toBeDefined();
+
+            // Create another assignment (with no DDAH)
+            // for position associated with other instructor
+            const otherAssignment = {
+                note: "",
+                applicant_id: databaseSeeder.seededData.applicant.id,
+                position_id: otherPosition.id,
+                start_date: "2019-09-02T00:00:00.000Z",
+                end_date: "2019-12-31T00:00:00.000Z",
+            };
+            resp = await apiPOST("/admin/assignments", otherAssignment);
+            expect(resp).toHaveStatus("success");
+            foreignAssignment = resp.payload;
+        });
 
-        // Get newly created instructor and create a DDAH for them
-        resp = await apiGET(`/admin/instructors`);
-        expect(resp).toHaveStatus("success");
-        const instructorId = resp.payload.find(
-            (instructor) => instructor.utorid === instructorUser.utorid
-        )?.id;
-        expect(instructorId).toBeDefined();
-        const newDDAH = await createDdahWithFixedDuties(instructorId);
+        beforeEach(async () => {
+            await switchToInstructorOnlyUser();
+        });
```

Comment:
I like how you are making sure your active user is instructor only before each test!